packages/file/lib/src/backends/memory/memory_file.dart

```diff
@@ -173,8 +174,13 @@ class MemoryFile extends MemoryFileSystemEntity implements File {
       openSync(mode: mode);
 
   @override
-  io.RandomAccessFile openSync({io.FileMode mode = io.FileMode.read}) =>
-      throw UnimplementedError('TODO');
+  io.RandomAccessFile openSync({io.FileMode mode = io.FileMode.read}) {
+    if (utils.isWriteMode(mode) && !existsSync()) {
+      createSync();
+    }
```

Comment:
Don't we need to throw an exception if you open the file for reading and it doesn't exist?

Reply:
Accessing `resolveBacking` already checks for file existence and will throw a `FileSystemException` if no file exists. (This the reason why `createSync` is handled in `MemoryFile.openSync` instead of in `MemoryRandomAccessFile`'s constructor.)